### PR TITLE
Add petsathome_gb + vets4pets_gb (850 sites)

### PR DESCRIPTION
**Pull request by mjoe999**
Something for everyone here, the commit is self contained but at a later stage consider:

@iandees : see the get_locations() method, sugar for locations/searchable_points, a common helper?

@Cj-Malone: I have had to frig the LD in each spider prior to parsing. I guess something to keep an eye on i.e. if we come across pattern too many times then support in library code?

Note: both questions are forward looking ... not for this commit I think.

### CODE REVIEW

File: locations/spiders/petsathome_gb.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        for key in keys:
+            if row.get(key):
+                return row[key]
+        return None
+
+    if area_field_filter and type(area_field_filter) is not list:
+        area_field_filter = [area_field_filter]
+    with open("./locations/searchable_points/{}".format(areas_csv_file)) as points:
+        for row in csv.DictReader(points):
+            lat = get_key(row, ["latitude", "lat"])
```
Comment (Cj-Malone) [inline locations/spiders/petsathome_gb.py:24]: I think `get_key` here is an over complication. I like the idea in principle, but here we control the source data. We should just unify on `latitude` or `lat`.

`head -n1 locations/searchable_points/*` shows them all using `latitude,longitude` except for `germany_centroids_80km_radius_country.csv` and `germany_grid_15km.csv`. Lets fix that, and make sure to update any spiders that use them.

File: locations/spiders/petsathome_gb.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import csv
+from locations.linked_data_parser import LinkedDataParser
+from scrapy.http import FormRequest
+
+
+# This method could be moved into library code and documented, tested etc, supports calls of the form:
+# get_locations("eu_centroids_40km_radius_country.csv")
+# get_locations("eu_centroids_40km_radius_country.csv", ["GB", "IE"])
+# get_locations("us_centroids_50mile_radius_state.csv", "NY")
+def get_locations(areas_csv_file, area_field_filter=None):
```
Comment (Cj-Malone) [inline locations/spiders/petsathome_gb.py:13]: In spider this seems like overkill, but if it ends up shared code, I can understand it,

File: locations/spiders/vets4pets_gb.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    def parse_func(self, response):
+        ld = LinkedDataParser.find_linked_data(response, "VeterinaryCare")
+        if ld:
+            # Should we support lists of addresses in the common parser code and take the first?
+            ld["address"] = ld["address"][0]
+            item = LinkedDataParser.parse_ld(ld)
+            item["ref"] = response.url
+            extract_google_position(item, response)
+            if "petsathome" in item["street_address"].lower().replace(" ", ""):
+                self.logger.info("%s located in Pets at Home store", item["name"])
```
Comment (Cj-Malone) [inline locations/spiders/vets4pets_gb.py:43]: Do we get anything from logging this?

File: locations/spiders/petsathome_gb.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    def parse_func(self, response):
+        for node in json.loads(response.body)["data"]:
+            if "groom-room" not in node["slug"]:
+                yield scrapy.Request("https://community.petsathome.com" + node["slug"])
+
+    def parse(self, response):
+        ld = LinkedDataParser.find_linked_data(response, "PetStore")
+        if ld:
+            # Our LD parser is reasonably strict, help petaathome meet its requirements.
+            ld["address"]["@type"] = "PostalAddress"
```
Comment (Cj-Malone) [inline locations/spiders/petsathome_gb.py:74]: I think it's reasonable for us to assume any `address` dict is a `@type=PostalAddress`.

[schema.org](https://schema.org/address) shows only `PostalAddress` and `Text` as expected types. Update the parser, add a comment that we're assuming PostalAddress and add a test.

File: locations/spiders/vets4pets_gb.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        Rule(LinkExtractor(allow="/practices/"), callback="parse_func", follow=True)
+    ]
+    allowed_domains = ["vets4pets.com"]
+    download_delay = 0.2
+
+    def parse_func(self, response):
+        ld = LinkedDataParser.find_linked_data(response, "VeterinaryCare")
+        if ld:
+            # Should we support lists of addresses in the common parser code and take the first?
+            ld["address"] = ld["address"][0]
```
Comment (Cj-Malone) [inline locations/spiders/vets4pets_gb.py:38]: I don't know if it's [to spec], but it seems reasonable to use the first in an array. I'd vote for changing the parser.

Comment (Cj-Malone) [reply locations/spiders/vets4pets_gb.py:38]: https://schema.org/Organization

File: locations/spiders/petsathome_gb.py (hunk 1 above)
Comment (mjoe999) [reply locations/spiders/petsathome_gb.py:24]: I would propose that we have this discussion and changes in a subsequent PR that pulled this method out. I don't want to do too much in a single change.

File: locations/spiders/vets4pets_gb.py (hunk 5 above)
Comment (mjoe999) [reply locations/spiders/vets4pets_gb.py:38]: I will do this.

File: locations/spiders/petsathome_gb.py (hunk 4 above)
Comment (mjoe999) [reply locations/spiders/petsathome_gb.py:74]: I will do this.

File: locations/spiders/petsathome_gb.py (hunk 1 above)
Comment (Cj-Malone) [reply locations/spiders/petsathome_gb.py:24]: Will be fixed with #3796

File: locations/spiders/vets4pets_gb.py (hunk 3 above)
Comment (mjoe999) [reply locations/spiders/vets4pets_gb.py:43]: Removing

**Comment by mjoe999** [comment]
All changes now made based on review.

File: locations/spiders/vets4pets_gb.py [hunk 6]
```diff
@@ -34,12 +34,9 @@ class Vets4PetsGBSpider(CrawlSpider):
     def parse_func(self, response):
         ld = LinkedDataParser.find_linked_data(response, "VeterinaryCare")
```
Comment (Cj-Malone) [inline locations/spiders/vets4pets_gb.py:35]: This can become a LinkedDataParser.parse

Comment (mjoe999) [reply locations/spiders/vets4pets_gb.py]: Yes

**Review by Cj-Malone — APPROVED** [review]
LGTM

**Pull request merged**